pkg/apis/testing/v1alpha1/testdefinition_types.go

```diff
@@ -0,0 +1,74 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestDefinitionSpec defines the desired state of TestDefinition
+type TestDefinitionSpec struct {
+	PodSpec v1.PodSpec `json:"pod_spec"`
+	// If there are some problems with given test, we add possibility to don't execute them
+	// On Testsuite level such test should be marked as a skipped
+	Skip bool `json:"skip,omitempty"`
+	// If test is working on data that can be modified by another test,
+	// I would like to run it in separation.
+	DisableConcurrency bool `json:"disable_concurrency,omitempty"`
+	// List of components this test depends on.
+	Components []string `json:"components,omitempty"`
+	// Test should be interrupted after the timeout.
+	// On test suite level such test should be marked as a timeouted
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+}
+
+// TestDefinitionStatus defines the observed state of TestDefinition
+type TestDefinitionStatus struct {
```

Comment:
IMO if we do not have a status then this field should be removed

Reply:
Agreed. ConfigMap for example doesn't have status.